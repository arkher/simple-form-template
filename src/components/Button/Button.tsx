import React from "react";

interface IButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  isLoading?: boolean;
  loadingText?: string;
}
const Button = ({ isLoading, loadingText, disabled, ...props }: IButton) => {
  return (
    <button
      {...props}
      className="absolute bottom-4 left-4 right-4 w-90 h-12 flex border-solid border-1 hover:border-gray-500 focus:border-gray-500 focus:outline-none border-gray-400 justify-center items-center bg-gray-200 text-black rounded-md py-2 px-12 md:w-56 md:relative md:left-0 md:bottom-0 disabled:hover:border-gray-400 disabled:cursor-not-allowed disabled:bg-gray-200  disabled:border-gray-400 disabled:text-gray-500"
      disabled={isLoading || disabled}
      data-testid="button"
    >
      {isLoading ? <>{loadingText || "Processando..."}</> : props.children}
    </button>
  );
};

export default Button;
