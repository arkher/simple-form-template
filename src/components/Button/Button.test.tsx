// Button
import { fireEvent, render, screen } from "@testing-library/react";

import Button from "./Button";

describe("Button", () => {
  it("should render the button", () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText("Click me")).toBeTruthy();
  });

  it("should call the onClick function when clicked", () => {
    const onClick = jest.fn();
    render(
      <Button data-testid="button" onClick={onClick}>
        Click me
      </Button>
    );
    fireEvent.click(screen.getByText("Click me"));
    expect(onClick).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    render(<Button>Click me</Button>);
    expect(screen.getByTestId("button")).toMatchSnapshot();
  });
});
