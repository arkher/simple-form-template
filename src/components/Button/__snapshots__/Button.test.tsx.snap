// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  class="absolute bottom-4 left-4 right-4 w-90 h-12 flex border-solid border-1 hover:border-gray-500 focus:border-gray-500 focus:outline-none border-gray-400 justify-center items-center bg-gray-200 text-black rounded-md py-2 px-12 md:w-56 md:relative md:left-0 md:bottom-0 disabled:hover:border-gray-400 disabled:cursor-not-allowed disabled:bg-gray-200  disabled:border-gray-400 disabled:text-gray-500"
  data-testid="button"
>
  Click me
</button>
`;
